rotkehlchen/globaldb/upgrades/v12_v13.py

```diff
@@ -0,0 +1,90 @@
+import csv
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from rotkehlchen.logging import enter_exit_debug_log
+from rotkehlchen.utils.progress import perform_globaldb_upgrade_steps, progress_step
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.drivers.gevent import DBConnection, DBCursor
+    from rotkehlchen.db.upgrade_manager import DBUpgradeProgressHandler
+
+
+@enter_exit_debug_log(name='globaldb v12->v13 upgrade')
+def migrate_to_v13(connection: 'DBConnection', progress_handler: 'DBUpgradeProgressHandler') -> None:  # noqa: E501
+    """This globalDB upgrade does the following:
+    - Add new token kinds (SPL Tokens & MetaPlex NFTs) for the solana ecosystem.
+    - Add solana_tokens table and migrate solana tokens to that.
+
+    This upgrade takes place in v1.40.0"""
+    @progress_step('Add new token kinds for Solana')
+    def _update_token_kinds_to_include_solana(write_cursor: 'DBCursor') -> None:
+        write_cursor.executescript("""
+            INSERT OR IGNORE INTO token_kinds(token_kind, seq) VALUES ('D', 4);
+            INSERT OR IGNORE INTO token_kinds(token_kind, seq) VALUES ('E', 5);
+        """)
+
+    @progress_step('Add solana tokens table and populate from CSV')
+    def _add_and_populate_solana_tokens(write_cursor: 'DBCursor') -> None:
+        """Create solana tokens table and update asset identifiers to use new solana format"""
+        write_cursor.execute("""
+        CREATE TABLE IF NOT EXISTS solana_tokens (
+            identifier TEXT PRIMARY KEY NOT NULL COLLATE BINARY,
+            token_kind CHAR(1) NOT NULL DEFAULT('D') REFERENCES token_kinds(token_kind),
+            address VARCHAR[44] NOT NULL,
+            decimals INTEGER,
+            protocol TEXT,
+            FOREIGN KEY(identifier) REFERENCES assets(identifier) ON UPDATE CASCADE ON DELETE CASCADE
+        )""")  # noqa: E501
+
+        dir_path = Path(__file__).resolve().parent.parent.parent
+        csv_file = dir_path / 'data' / 'solana_tokens_data.csv'
+        if not csv_file.exists():
+            # log here.
+            return
+
+        asset_updates = []
+        common_asset_updates = []
+        solana_tokens_data = []
+        multiasset_updates = []
+        user_owned_updates = []
+        price_from_updates = []
+        price_to_updates = []
+        binance_base_updates = []
+        binance_quote_updates = []
+        with csv_file.open(encoding='utf-8') as f:
+            reader = csv.DictReader(f)
+            for row in reader:
+                # skip duplicate mapping; TRISIG maps to same identifier as TRISG
+                if (old_id := row['old_id']) == 'TRISIG':
+                    continue
+
+                asset_updates.append(((new_id := f'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/spl_token:{row["address"]}'), old_id))  # noqa: E501
```

Comment:
```suggestion
                asset_updates.append(((new_id := f'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/token:{row["address"]}'), old_id))  # noqa: E501
```
From the example here: https://namespaces.chainagnostic.org/solana/caip19 I think this should just be `token` rather than `spl_token`